nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py

```diff
@@ -494,9 +503,42 @@ def setup_optimizer_param_groups(self):
         else:
             self._optimizer_param_groups = get_params_for_weight_decay_optimization(self.model)
 
+    def setup_mcore_distributed_parallel(self):
+        """Set up mcore distributed data parallel """
+        if self.with_distributed_adam and self.use_mcore_dist_optim:
+            config = get_model_config(self.model[0])
+            ddp_config = DistributedDataParallelConfig(
+                grad_reduce_in_fp32=self.megatron_amp_O2,
+                overlap_grad_reduce=self.cfg.optim.get('mcore_overlap_grad_sync', False),
+                use_distributed_optimizer=True,
+                check_for_nan_in_grad=False,
+                # mcore bucket_size is based on num of parameters, therefore not
+                # using grad_allreduce_chunk_size_mb to configure bucket_size here
+                bucket_size=self.cfg.optim.get('mcore_ddp_bucket_size', None),
```

Comment:
Thanks @erhoo82 for the comments. For this flag, the reason to use a new option bucket_size instead of sizes in mb is try to avoid confusion - 
In Mcore, regardless of dtype, different param groups are assigned the same bucket size (the num of parameters): https://gitlab-master.nvidia.com/ADLR/megatron-lm/-/blob/main/megatron/core/distributed/distributed_data_parallel.py?ref_type=heads#L72 . However in NeMo the actually bucket_size is calculated based on dtype https://github.com/NVIDIA/apex/blob/master/apex/contrib/optimizers/distributed_fused_adam.py#L700 

If we use similar `_mb` annotation here, we need to pre-assume a certain dtype for calculating bucket_size. Setting mcore in this way might be a bit confusing.
But I do agree using fixed bucket_size regardless of dtype is more nature in term of communication. Do you think it's better to change the Mcore side to make each bucket the same size in mb and keep NeMo interface. Otherwise I would prefer to use a different option if bucket_size is based on num of parameters.